fdbclient/AuditUtils.actor.cpp

```diff
@@ -45,6 +49,87 @@ ACTOR static Future<std::vector<AuditStorageState>> getLatestAuditStatesImpl(Tra
 	return auditStates;
 }
 
+ACTOR Future<Void> clearAuditMetadata(Database cx, AuditType auditType, UID auditId, bool clearProgressMetadata) {
+	state Transaction tr(cx);
+	TraceEvent(SevDebug, "ClearAuditMetadataStart", auditId).detail("AuditKey", auditKey(auditType, auditId));
+
+	loop {
+		try {
+			tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+			tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			// Clear audit
+			tr.clear(auditKey(auditType, auditId));
+			if (clearProgressMetadata) {
+				// Here, we do not distinguish which range exactly used by an audit
+				// Simply clear any possible place of storing the progress of the audit
+				tr.clear(auditServerBasedProgressRangeFor(auditType, auditId));
```

Comment:
Based on the type, we know which key range we should clear?

Reply:
Yes. The progress metadata prefix (so range) is decided given an auditType and an auditId. See: auditServerBasedProgressRangeFor and auditRangeBasedProgressRangeFor